src/transformers/tokenization_utils_fast.py

```diff
@@ -893,3 +894,37 @@ def train_new_from_iterator(
             kwargs["additional_special_tokens"] = additional_special_tokens
 
         return self.__class__(tokenizer_object=tokenizer, **kwargs)
+
+
+def convert_tiktoken_to_fast(encoding: Any, output_dir: str):
```

Comment:
very nice! 
We should do :
```python
from tiktoken import get_encoding

# You can load your custom encoding or the one provided by OpenAI
encoding = get_encoding("gpt2")
```
in this function directly IMO! 

Also let's maybe place this under the `integration` folder ! 🤗

Reply:
I wouldn't like to load the encoding inside the function to allow for custom encodings like https://github.com/openai/tiktoken?tab=readme-ov-file#extending-tiktoken 

We could allow `Encoding | str` as input and load the encoding with `get_encoding` if a str is passed.

Reply:
Moved it to `integration/tiktoken.py`